src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -209,18 +229,23 @@ private TReport RecordDiagnosticReport(Document document, VSDiagnostic[] diagnos
             }
         }
 
-        private static VSDiagnostic ConvertDiagnostic(Document document, SourceText text, DiagnosticData diagnosticData)
+        private VSDiagnostic ConvertDiagnostic(Document document, SourceText text, DiagnosticData diagnosticData)
         {
             Contract.ThrowIfNull(diagnosticData.Message, $"Got a document diagnostic that did not have a {nameof(diagnosticData.Message)}");
             Contract.ThrowIfNull(diagnosticData.DataLocation, $"Got a document diagnostic that did not have a {nameof(diagnosticData.DataLocation)}");
 
             var project = document.Project;
+
+            // Razor wants to handle all span mapping themselves.  So if we are in razor, return the raw doc spans, and
```

Comment:
So unfortunately this is only true if we're using `pull` diagnostics. If for some reason Razor is in a "push" mode then you'd still need to do the remapping because we're nowhere inbetween to middlelayer those requests.

Reply:
oh, good to know!  i will revert hte push part.

Reply:
done.